pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -1098,59 +1150,131 @@ func (r *RepoSyncReconciler) mutationsFor(ctx context.Context, rs *v1beta1.RepoS
+	if err := cl.List(ctx, &cmList, &client.ListOptions{Namespace: configsync.ControllerNamespace}); err != nil {
+		return nil, fmt.Errorf("failed to list ConfigMaps in %s namespace: %w", configsync.ControllerNamespace, err)
+	}
+	for _, cm := range cmList.Items {
+		if !strings.HasPrefix(cm.Name, core.NsReconcilerPrefix) {
+			continue
+		}
+		annotations := cm.GetAnnotations()
+		if value, found := annotations[repoSyncNameAnnotationKey]; !found || value != rs.Name {
+			// this ConfigMap was not created for this RepoSync
+			continue
+		}
+		if value, found := annotations[repoSyncNamespaceAnnotationKey]; !found || value != rs.Namespace {
+			// this ConfigMap was not created for this RepoSync
+			continue
+		}
+		if _, found := copiedConfigMapNames[cm.Name]; found {
+			// this ConfigMap is still needed for this RepoSync
+			continue
+		}
+		// none of the above conditions apply, so this ConfigMap is no longer needed, and we should delete it for
+		// cleanup
+		if err := cl.Delete(ctx, &cm); err != nil {
+			return nil, fmt.Errorf("failed to delete ConfigMap: %w", err)
+		}
 	}
 
 	return newValuesFileRefs, nil
 }
 
-func copyOneConfigMapToCms(ctx context.Context, cl client.Client, objRef types.NamespacedName) (string, error) {
-	var cm corev1.ConfigMap
-	if err := cl.Get(ctx, objRef, &cm); err != nil {
-		return "", fmt.Errorf("failed to get referenced ConfigMap: %w", err)
+func copyOneConfigMapToCms(ctx context.Context, cl client.Client, objRef types.NamespacedName, rs *v1beta1.RepoSync) (string, error) {
+	var sourceConfigMap corev1.ConfigMap
+	err := cl.Get(ctx, objRef, &sourceConfigMap)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// the source ConfigMap doesn't exist, so there is nothing to copy. The previous validation checks
+			// will already surface this as an error on the RepoSync, so we don't need to do anything here.
+			return "", nil
+		}
+		return "", fmt.Errorf("unexpected error fetching ConfigMap: %w", err)
+
+	}
+
+	sourceConfigMapCopy, err := createSourceConfigMapCopy(rs, sourceConfigMap)
+	if err != nil {
+		return "", fmt.Errorf("failed to create source ConfigMap copy: %w", err)
 	}
 
+	var existingConfigMapCopy corev1.ConfigMap
+	err = cl.Get(ctx, types.NamespacedName{Name: sourceConfigMapCopy.Name, Namespace: sourceConfigMapCopy.Namespace}, &existingConfigMapCopy)
+	if err == nil {
+		// the copied ConfigMap already exists, and is immutable, so we don't need to do anything.
+		return sourceConfigMapCopy.Name, nil
+	}
+	if err != nil && !apierrors.IsNotFound(err) {
+		return "", fmt.Errorf("unexpected error fetching configmap: %w", err)
+	}
+
+	if err := cl.Create(ctx, &sourceConfigMapCopy); err != nil {
+		return "", fmt.Errorf("failed to create copy of configmap: %w", err)
+	}
+	return sourceConfigMapCopy.Name, nil
+}
+
+func createSourceConfigMapCopy(rs *v1beta1.RepoSync, sourceConfigMap corev1.ConfigMap) (corev1.ConfigMap, error) {
+	if rs.Namespace == configsync.ControllerNamespace {
```

Comment:
nit: add a comment that says this is guarded by validation code and shouldn't happen, and we're just doing a sanity check here.

Reply:
done

Reply:
Hmm I didn't see the change (same for other comments that aren't marked resolved). Have you pushed the change?

Reply:
🤔 Should be there in the [last commit](https://github.com/GoogleContainerTools/kpt-config-sync/pull/752/commits/43fd38a1066665400ff96487b9e3d3a7f4f5e97c). 

Do you see the "address nits" commit here? https://github.com/GoogleContainerTools/kpt-config-sync/pull/752/commits